src/components/Signup/Signup.js

```diff
@@ -1,5 +1,133 @@
-import React from "react";
-
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { signUpHandler } from "../../ApiServices/ApiServices";
 export default function Signup() {
-  return <div>Signup</div>;
+  const [signIn, setSignin] = useState({
+    email: "",
+    firstName: "",
+    lastName: "",
+    password: "",
+  });
+  const navigateTo = useNavigate();
+  const signUp = async (signIn) => {
+    const { status } = await signUpHandler(signIn);
+    if (status === 201) {
+      navigateTo("/login");
+    }
+  };
+  return (
+    <div className="d-flex flex-justify-center align-items-center">
+      <div className="card nav-yellow-shadow  login-container flex-column">
+        <div className="d-flex flex-justify-center">
+          <h3 className="white-text-color">Signup</h3>
+        </div>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            signUp(signIn);
+          }}
+        >
+          <div className=" mt-2 ml-1 d-flex flex-column">
+            <label htmlFor="input-fname" className="ml-2 white-text-color">
+              First Name
+            </label>
+            <input
+              className="input input-login black-light-bg white-text-color"
+              id="input-fname"
+              placeholder="Loreum"
+              onChange={(e) => {
+                let timer = setTimeout(() => {
+                  setSignin({ ...signIn, firstName: e.target.value });
+                  clearTimeout(timer);
+                }, 1000);
+              }}
+            />
+          </div>
+          <div className=" mt-2 ml-1 d-flex flex-column">
+            <label htmlFor="input-lname" className="ml-2 white-text-color">
+              Last Name
+            </label>
+            <input
+              className="input input-login black-light-bg white-text-color"
+              id="input-lname"
+              placeholder="Ipsum"
+              onChange={(e) => {
+                let timer = setTimeout(() => {
+                  setSignin({ ...signIn, lastName: e.target.value });
+                  clearTimeout(timer);
+                }, 1000);
+              }}
+            />
+          </div>
+          <div className=" mt-2 ml-1 d-flex flex-column">
+            <label htmlFor="input-email" className="ml-2 white-text-color">
+              Email
+            </label>
+            <input
+              className="input input-login black-light-bg white-text-color"
+              id="input-email"
+              placeholder="you@dreamstore.com"
+              type="email"
+              onChange={(e) => {
+                let timer = setTimeout(() => {
+                  setSignin({ ...signIn, email: e.target.value });
+                  clearTimeout(timer);
+                }, 1000);
```

Comment:
Didn't get the purpose of setting the data within a timeout action . Was there any issue when doing it without timeout

Reply:
Timer is used to reduce number of calls when we type input